CONTRIBUTING.md

```diff
@@ -1,106 +1,186 @@
+
+1.  Open any solution, eg "SDKs\Compute\Microsoft.Azure.Management.Compute.sln"
+2.  Build solution from Visual Studio
+
+### Full Build from command line
+
+1.  Open VS 2019 command prompt
+2.  Navigate to repository root directory
+3.  Invoke `msbuild build.proj /t:Build`
+
+### Single Service from Command Line
+
+1. Open VS 2019 command Prompt
+2. From the root directory
+3. Invoke `msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
 
-### To run tests:
-Using Visual Studio:
-  - Build project in VS
-  - "Test Explorer" window will get populated with tests. Right click on a test that you wish to Run/Debug.
+> _Build_ without any scope will build all management SDK's.
 
-Using the command line:
-msbuild .\build.proj /t:Runtests /p:Scope=Compute
-in the above example RunTests will build and run tests for Compute only
-or you can use command line CLI
-dotnet test Compute\Microsoft.Azure.Management.Compute\tests\Microsoft.Azure.Management.Tests.csproj
+### Create single nuget package
 
-#### Create single nuget package
 In order to build one package and run it's test
 `msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
 Nuget package will be created in root directory under \artifacts\packages\Debug (default configuration is Debug)
 
+## TO TEST:
+
+### Using Visual Studio:
+
+1. Build project in Visual Studio.
+2. **Test Explorer** window will get populated with tests. Select test and `Run` or `Debug`
+
+### Using the command line:
+
+Run e.g. `msbuild .\build.proj /t:"Runtests" /p:Scope=Compute`
+In the above example _RunTests_ will build and run tests for Compute only or you can use command line CLI
+`dotnet test Compute\Microsoft.Azure.Management.Compute\tests\Microsoft.Azure.Management.Tests.csproj`
+
 ### Non-Windows command line build
-Now you can use the same command on non-windows as above
-for e.g. on Ubuntu you can do something like below:
-`dotnet msbuild build.proj /t:Build /p:scope=Compute`
-`dotnet msbuild build.proj /t:RunTests /p:scope=Compute`
-`dotnet msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
+
+Now you can use the same command on non-windows as above for e.g. on Ubuntu you can do something like below:
+
+- `dotnet msbuild build.proj /t:Build /p:scope=Compute`
+- `dotnet msbuild build.proj /t:RunTests /p:scope=Compute`
+- `dotnet msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
 
 ### Update build tools
-Build tools are now downloaded as part of a nuget package under root\restoredPackages\microsoft.internal.netsdkbuild.mgmt.tools
-If for any reason there is an update to the build tools, you will then need to first delete directory root\restoredPackages\microsoft.internal.netsdkbuild.mgmt.tools and reexecute your build command. This will simply get the latest version of build tools.
 
+Build tools are now downloaded as part of a nuget package under `root\restoredPackages\microsoft.internal.netsdkbuild.mgmt.tools`
```

Comment:
@shahabhijeet once we correctly version and publish this package these steps will no longer be necessary.

Reply:
That is correct.
The only thing that needs to be added is, for getting latest version, you will have to pull latest from upstream and then execute build targets.
But yes once we version and start publishing nuget pacakges to nuget feed, no need to delete anything.

Reply:
I don't think we need to add any steps telling people to pull from latest upstream. They will automatically have the latest tools if they are on the latest source.